daemon/rpcserver.go

```diff
@@ -580,23 +581,39 @@ func (r *rpcServer) WatchAddress(ctx context.Context,
 func (r *rpcServer) NewAddress(ctx context.Context,
 	in *lnrpc.NewAddressRequest) (*lnrpc.NewAddressResponse, error) {
 
-	// Translate the gRPC proto address type to the wallet controller's
-	// available address types.
-	var addrType lnwallet.AddressType
-	switch in.Type {
-	case lnrpc.NewAddressRequest_WITNESS_PUBKEY_HASH:
-		addrType = lnwallet.WitnessPubKey
-	case lnrpc.NewAddressRequest_NESTED_PUBKEY_HASH:
-		addrType = lnwallet.NestedWitnessPubKey
+	if len(in.SubmarineHash) == 0 {
+		// Translate the gRPC proto address type to the wallet controller's
+		// available address types.
+		var addrType lnwallet.AddressType
+		switch in.Type {
+		case lnrpc.NewAddressRequest_WITNESS_PUBKEY_HASH:
+			addrType = lnwallet.WitnessPubKey
+		case lnrpc.NewAddressRequest_NESTED_PUBKEY_HASH:
+			addrType = lnwallet.NestedWitnessPubKey
+		}
+
+		addr, err := r.server.cc.wallet.NewAddress(addrType, false)
+		if err != nil {
+			return nil, err
+		}
+
+		rpcsLog.Infof("[newaddress] addr=%v", addr.String())
+		return &lnrpc.NewAddressResponse{Address: addr.String()}, nil
 	}
 
-	addr, err := r.server.cc.wallet.NewAddress(addrType, false)
+	//Create a new submarine address and associated script
+	addr, script, swapperPubKey, err := submarine.NewAddress(
+		activeNetParams.Params,
+		r.server.cc.wallet.WalletController.(*btcwallet.BtcWallet).InternalWallet().ChainClient(),
+		r.server.cc.wallet.Cfg.Database,
```

Comment:
isn't it better that the submarine package will be initialized with the database on startup like other packages of lightning lib that needs bolt db? this is instead of passing it on demand from the RPC layer.